app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SingleLiveEvent.kt

```diff
@@ -0,0 +1,77 @@
+package openfoodfacts.github.scrachx.openfood.utils
+
+/*
+ *  Copyright 2017 Google Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import android.util.Log
+import androidx.annotation.MainThread
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * A lifecycle-aware observable that sends only new updates after subscription, used for events like
+ * navigation and Snackbar messages.
+ *
+ *
+ * This avoids a common problem with events: on configuration change (like rotation) an update
+ * can be emitted if the observer is active. This LiveData only calls the observable if there's an
+ * explicit call to setValue() or call().
+ *
+ *
+ * Note that only one observer is going to be notified of changes.
+ */
+class SingleLiveEvent<T> : MutableLiveData<T>() {
```

Comment:
right, but I don't want to use it here because
- this VM already uses LiveData, SingleLiveEvent is a common approach for such cases with LiveData inside (actually I copied it from google with copyrights), so it will be right to say that we solve the problem without introducing new dependencies
- I don't want to use EventBus here because right now we have direct relation between activity and view model, and activity just reads the value from it. With introducing EventBus it will be harder relation like view model sends some data to EventBus, activity listen to the event of some type from EventBus, so in the result, we introduced new logical instance here and it becomes more complicated
- EventBus is quite hard to support imo

Reply:
> EventBus is quite hard to support imo

Why?

Reply:
because over time it became like an octopus with many tentacles and exists everywhere, it is hard to understand which class uses a particular event from EventBus, and changing some part of logic in one place can lead us to errors in unexpected places
and also you have to always control for subscribing\unsubscribing from it